src/Form/Form.js

```diff
@@ -173,6 +173,12 @@ class Form {
             })
             if (probableField?.value) {
                 formValues.credentials.username = probableField.value
+            } else if (
+                // If a form has phone + password(s) fields, save the phone as username
+                formValues.identities.phone &&
+                this.inputs.all.size - this.inputs.unknown.size < 4
+            ) {
+                formValues.credentials.username = formValues.identities.phone
```

Comment:
Just a quality of life improvements for those sites that use the phone number as a username. Here we try to save it as a username. I'm not doing anything on the autofill-side because of the risk for false matches, but this should give us a significant improvement by itself, as the signin forms of this type I've encountered were all working nicely and it certainly isn't any worse than it was before.